packages/aws-cdk-lib/aws-cloudfront-origins/test/function-url-origin.test.ts

```diff
@@ -41,3 +46,253 @@ test('Correctly renders the origin for a Lambda Function URL', () => {
+    keepaliveTimeout: cdk.Duration.seconds(60),
+  });
+
+  const originBindConfig = origin.bind(stack, { originId: 'StackOriginLambdaFunctionURL' });
+
+  expect(stack.resolve(originBindConfig.originProperty)).toMatchObject({
+    customOriginConfig: {
+      originReadTimeout: 120,
+      originKeepaliveTimeout: 60,
+    },
+  });
+});
+
+describe('FunctionUrlOriginAccessControl', () => {
+  test('Correctly adds permission to Lambda for CloudFront', () => {
+    const fn = new lambda.Function(stack, 'MyFunction', {
+      code: lambda.Code.fromInline('exports.handler = async () => {};'),
+      handler: 'index.handler',
+      runtime: lambda.Runtime.NODEJS_20_X,
+    });
+
+    const fnUrl = fn.addFunctionUrl({
+      authType: lambda.FunctionUrlAuthType.AWS_IAM,
+    });
+
+    new cloudfront.Distribution(stack, 'MyDistribution', {
+      defaultBehavior: {
+        origin: FunctionUrlOrigin.withOriginAccessControl(fnUrl, {}),
+      },
+    });
+
+    const template = Template.fromStack(stack);
+
+    template.hasResourceProperties('AWS::CloudFront::Distribution', {
+      DistributionConfig: {
+        Origins: Match.arrayWith([
+          Match.objectLike({
+            DomainName: {
+              'Fn::Select': [
+                2,
+                {
+                  'Fn::Split': [
+                    '/',
+                    {
+                      'Fn::GetAtt': ['MyFunctionFunctionUrlFF6DE78C', 'FunctionUrl'],
+                    },
+                  ],
+                },
+              ],
+            },
+            OriginAccessControlId: Match.objectLike({
+              'Fn::GetAtt': [
+                'MyDistributionOrigin1FunctionUrlOriginAccessControl0591AF75',
+                'Id',
+              ],
+            }),
+          }),
+        ]),
+      },
+    });
+
+    template.hasResourceProperties('AWS::Lambda::Permission', {
+      Action: 'lambda:InvokeFunctionUrl',
+      FunctionName: {
+        'Fn::GetAtt': ['MyFunctionFunctionUrlFF6DE78C', 'FunctionArn'],
+      },
+      Principal: 'cloudfront.amazonaws.com',
+      SourceArn: {
+        'Fn::Join': [
+          '',
+          [
+            'arn:',
+            { Ref: 'AWS::Partition' },
+            ':cloudfront::',
+            { Ref: 'AWS::AccountId' },
+            ':distribution/',
+            { Ref: 'MyDistribution6271DFB5' },
+          ],
+        ],
+      },
+    });
+  });
+
+  test('Correctly configures CloudFront Distribution with Origin Access Control', () => {
+    const fn = new lambda.Function(stack, 'MyFunction', {
+      code: lambda.Code.fromInline('exports.handler = async () => {};'),
+      handler: 'index.handler',
+      runtime: lambda.Runtime.NODEJS_20_X,
+    });
+
+    const fnUrl = fn.addFunctionUrl({
+      authType: lambda.FunctionUrlAuthType.AWS_IAM,
+    });
+
+    new cloudfront.Distribution(stack, 'MyDistribution', {
+      defaultBehavior: {
+        origin: FunctionUrlOrigin.withOriginAccessControl(fnUrl, {
```

Comment:
can we also have a test case that accepts only fnUrl?

```
FunctionUrlOrigin.withOriginAccessControl(fnUrl)
```

Reply:
Also a test case for OAC with default origin properties and with customOriginConfig properties
```
originSslProtocols
originProtocolPolicy
originReadTimeout
originKeepaliveTimeout
```

Reply:
fixed https://github.com/aws/aws-cdk/pull/31339/commits/d8b0b1669c37541c79acb94c8349991bc488e79c